From 5573782c815e3c4057e54b92a051ec9225d78d0a Mon Sep 17 00:00:00 2001
From: Sarah Walker <sarah.walker@imgtec.com>
Date: Tue, 13 Jun 2023 15:48:00 +0100
Subject: [PATCH 17/17] arm64: dts: ti: k3-am62-main: Add GPU device node [DO
 NOT MERGE]

Add the Series AXE GPU node to the AM62 device tree.

Signed-off-by: Sarah Walker <sarah.walker@imgtec.com>
---
 arch/arm64/boot/dts/ti/k3-am62-main.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-am62-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62-main.dtsi
index b3e4857bbbe4..ad13414acf18 100644
--- a/arch/arm64/boot/dts/ti/k3-am62-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62-main.dtsi
@@ -892,4 +892,17 @@ mcasp2: audio-controller@2b20000 {
 		power-domains = <&k3_pds 192 TI_SCI_PD_EXCLUSIVE>;
 		status = "disabled";
 	};
+
+        gpu: gpu@fd00000 {
+                compatible = "ti,am62-gpu", "img,powervr-seriesaxe";
+                reg = <0 0x0fd00000 0 0x20000>;
+                power-domains = <&k3_pds 187 TI_SCI_PD_EXCLUSIVE>;
+                clocks = <&k3_clks 187 0>;
+                clock-names = "core";
+                interrupts = <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
+                interrupt-names = "gpu";
+                #cooling-cells = <2>;
+                #cooling-min-level = <0>;
+                #cooling-max-level = <3>;
+        };
 };
-- 
2.39.2

